### PR TITLE
Add Typo3 Theme

### PR DESCRIPTION
**Pull request by allcaps**
Thanks for contributing to Wagtail! 🎉

Before submitting, please review the contributor guidelines <https://docs.wagtail.io/en/latest/contributing/index.html> and check the following:

* Do the tests still pass? (https://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* For Python changes: Have you added tests to cover the new/fixed behaviour?
* For front-end changes: Did you test on all of Wagtail’s supported browsers? **Please list the exact versions you tested**.
* For new features: Has the documentation been updated accordingly?

### CODE REVIEW

**Comment by Stormheg** [comment]
Known issue: some automatically generated docs are wrapped in quote blocks. [example](https://wagtail--6792.org.readthedocs.build/en/6792/reference/contrib/routablepage.html#module-wagtail.contrib.routable_page.models)

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -e .[docs]
+          pip install sphinx-typo3-theme==4.4.3
```
Comment (thibaudcolas) [inline .github/workflows/docs.yml]: This dependency should be defined in setup.py, as well as `docs/requirements.txt`, and removed from here.

File: docs/_static/css/custom.css [hunk 2]
```diff
@@ -106,3 +106,206 @@ a.wagtailspace:hover svg {
...
+.wagtailspace .close {
+    margin-left: 1em;
+    margin-bottom: 1em;
+    cursor: pointer;
+    font-weight: bold;
+    font-size: 1.25em;
+}
+
+/* Bounce the UFO on hover  */
+@-webkit-keyframes bounce {
```
Comment (thibaudcolas) [inline docs/_static/css/custom.css:188]: We should remove this, no need to support Safari 8.

File: docs/_static/css/custom.css [hunk 3]
```diff
@@ -106,3 +106,206 @@ a.wagtailspace:hover svg {
...
+  0%, 100% {
+    transform: translateY(0);
+  }
+  50% {
+    transform: translateY(-5px);
+  }
+}
+
+a.wagtailspace:hover svg {
+  -webkit-animation-duration: .5s;
```
Comment (thibaudcolas) [inline docs/_static/css/custom.css:207]: Same for all of these `-webkit-` prefixed styles.

File: docs/_templates/layout.html [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+<head>
+    <meta charset="utf-8" />
+{%- if metatags %}
+    {{ metatags }}
+{%- endif %}
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="docsearch:name" content="{{ project|striptags|e }}" />
+    <meta name="docsearch:package_type" content="{{ package_type|striptags|e }}" />
+    <meta name="docsearch:release" content="{{ release|striptags|e }}" />
+    <meta name="docsearch:version" content="{{ version|striptags|e }}" />
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:39]: Does this already have support for Algolia DocSearch? 🤔

File: docs/_templates/footer.html [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                {%-
+                    if theme_project_home
+                    or theme_project_contact
+                    or theme_project_discussions
+                    or theme_project_issues
+                    or theme_project_repository
+                %}
+                <ul class="footer-simplemenu">
+                    {%- if theme_project_home %}
+                    <li><a href="{{ theme_project_home|e }}" title="Home"><span>Home</span></a></li>
```
Comment (thibaudcolas) [inline docs/_templates/footer.html:17]: Could we remove all those `title` attributes? Realise this isn’t used currently but really these shouldn’t be there.

File: docs/_templates/layout.html [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+    <link href="https://fonts.googleapis.com/css2?family=Roboto+Slab:wght@500&display=swap" rel="stylesheet">
+{%- for css in css_files %}
+  {%- if css|attr("filename") %}
+    {{ css_tag(css) }}
+  {%- else %}
+    <link rel="stylesheet" type="text/css" href="{{ pathto(css, 1) }}" />
+  {%- endif %}
+{%- endfor %}
+<!-- linktags -->
+{%- block linktags %}
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:65]: I believe all of these are useless.

File: docs/_templates/layout.html [hunk 7]
```diff
@@ -0,0 +1,308 @@
...
+  {%- endif %}
+  {%- if prev %}
+    <link rel="prev" title="{{ prev.title|striptags|e }}" href="{{ prev.link|e }}" />
+  {%- endif %}
+{% endblock -%}
+<!-- /linktags -->
+<!-- extrahead -->
+{%- block extrahead %}{% endblock -%}
+<!-- /extrahead -->
+    <script src="{{ pathto('_static/js/modernizr.min.js', 1) }}"></script>
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:93]: We should remove this.

File: docs/_templates/layout.html [hunk 8]
```diff
@@ -0,0 +1,308 @@
...
+{% if theme_docstypo3org -%}
+<!-- UNIVERSE BAR START -->
+    <script src="https://typo3.azureedge.net/typo3infrastructure/universe/dist/webcomponents-loader.js"></script>
+    <script type="module" src="https://typo3.azureedge.net/typo3infrastructure/universe/dist/typo3-universe.js"></script>
+<!-- UNIVERSE BAR END -->
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:98]: We should remove that as well.

File: docs/_templates/layout.html [hunk 9]
```diff
@@ -0,0 +1,308 @@
...
+    <div class="page">
+      {%- if theme_docstypo3org %}
+        <div class="page-topbar">
+            <div class="page-topbar-inner">
+                {%- block pagetopbar %}
+                <!-- UNIVERSE BAR START -->
+                <typo3-universe active="documentation">
+                    <div style="display: block; height: 44px; background-color: #313131;"></div>
+                </typo3-universe>
+                <!-- UNIVERSE BAR END -->
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:113]: To remove?

File: docs/_templates/layout.html [hunk 10]
```diff
@@ -0,0 +1,308 @@
...
+<!-- scripts2 -->
+    {%- block scripts2 %}{% endblock -%}
+<!-- /scripts2 -->
+<!-- footer2 -->
+    {%- block footer2 %}{% endblock -%}
+<!-- /footer2 -->
+  {%- if theme_docstypo3org %}
+    <script type="application/javascript" src="https://app.usercentrics.eu/latest/main.js" id="R9wBKTwzv"></script>
+    <script type="text/plain" data-usercentrics="Matomo" src="/js/piwik.js"></script>
+  {%- endif %}
```
Comment (thibaudcolas) [inline docs/_templates/layout.html:305]: To remove.

File: docs/conf.py [hunk 11]
```diff
@@ -140,7 +145,14 @@
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-# html_theme_options = {}
+html_theme_options = {
+    "logo_title": "Wagtail Documentation",
+    "logo_alt": "Wagtail logo",
+    "logo_url": "/",
+    "project_home": "https://wagtail.io/",
+    "project_repository": "https://github.com/wagtail/wagtail/"
```
Comment (thibaudcolas) [inline docs/conf.py:151]:
```suggestion
    "project_repository": "https://github.com/wagtail/wagtail"
```

File: docs/conf.py [hunk 12]
```diff
@@ -140,7 +145,14 @@
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-# html_theme_options = {}
+html_theme_options = {
+    "logo_title": "Wagtail Documentation",
+    "logo_alt": "Wagtail logo",
```
Comment (thibaudcolas) [inline docs/conf.py:148]:
```suggestion
    "logo_alt": "Wagtail",
```

File: docs/conf.py [hunk 13]
```diff
@@ -140,7 +145,14 @@
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-# html_theme_options = {}
+html_theme_options = {
+    "logo_title": "Wagtail Documentation",
```
Comment (thibaudcolas) [inline docs/conf.py:147]: Should be removed, `alt` is enough.

**Review by thibaudcolas — CHANGES_REQUESTED** [review]
Issues I spotted in the site’s output only:

- [ ] Edit on GitHub link is incorrect, "https://github.com/unused,%20deprecated/edit/unused,%20deprecated/Documentation/Localization.unused,%20deprecated/index.rst"
- [ ] Code snippets on https://wagtail--6792.org.readthedocs.build/en/6792/getting_started/index.html are impossible to read

File: docs/conf.py (hunk 13 above)
Comment (allcaps) [reply docs/conf.py]: I also removed this from HTML.

**Comment by kaedroho** [comment]
Thanks for this @allcaps!

Would it be possible to squash the commits, or split this into multiple PRs to make it easier to see what's going on?

**Comment by Stormheg** [comment]
@kaedroho I've extracted the content changes here: https://github.com/wagtail/wagtail/pull/6835

**Comment by Stormheg** [comment]
Since opening this pull request we have decided to create our own sphinx theme ([sphinx_wagtail_theme](https://github.com/wagtail/sphinx_wagtail_theme)) instead of trying to hack the typo3 theme to fit our needs - which is what this pull request is doing.

Content changes in this PR were merged seperately in https://github.com/wagtail/wagtail/pull/6835

Going to close this in favor of https://github.com/wagtail/wagtail/pull/6930

**Pull request closed without merging**